src/components/dropdown.js

```diff
@@ -0,0 +1,146 @@
+      .then(response => {
+        const data = {}
+        for (const record of response.data) {
+          data[record.id] = record;
+        }
+        this.setState({
+          users: data,
+        });
+      })
+      .catch(function (error) {
+        console.log(error);
+      });
+  }
+
+  toggleDropdown() {
+    this.setState(prevState => ({
+      isDropdownVisible: !prevState.isDropdownVisible,
+      checkedUsers: [],
+    }));
+  }
+
+  getCheckedUsers = event => {
+    let newSelectedUsers = this.state.checkedUsers;
+    let isChecked = event.target.checked;
+    let userId = parseInt(event.target.value);
+    this.setState({
+      checkedUsers: isChecked ? newSelectedUsers.concat(userId) : this.state.checkedUsers.filter(user => user !== userId),
+    });
+  }
+
+  searchUser = (event) => {
+    this.setState({
+      isDropdownVisible: true,
+      searchValue: event.target.value,
+    });
+  }
+
+  deleteUser = (event, userId) => {
+    this.setState({
+      selectedUsers: this.state.selectedUsers.filter(user => user !== userId),
+    });
+  }
+
+  deleteAllUsers() {
+    this.setState({
+      selectedUsers: [],
+    })
+  }
+
+  submitHandler() {
+    const { selectedUsers, checkedUsers } = this.state;
+    this.setState({
+      isDropdownVisible: checkedUsers.length > 0 ? false : true,
+      isSubmitted: true,
+      selectedUsers: [...selectedUsers, ...checkedUsers],
+      checkedUsers: [],
+      searchValue: '',
+    });
+  }
+
+  transformDropdownUsers = () => {
+    const { searchValue, selectedUsers, users } = this.state;
+    let updatedList = Object.values(users).filter(user => !selectedUsers.includes(user.id));
+    if (searchValue) {
+      updatedList = updatedList.filter(user => user.username.toLowerCase().search(searchValue.toLowerCase()) !== -1);
+      updatedList = updatedList.map(user => {
+        let modifiedUsername = SearchHighlight(user.username, searchValue);
+        return { ...user, username: modifiedUsername };
+      });
+    }
+    return updatedList;
+  }
+
+  closeDropdownOutsideClick = (event) => {
+    const { checkedUsers } = this.state;
+    if (event.target.contains(this.dropdown)) {
+      this.setState({
+        isDropdownVisible: checkedUsers.length > 0 ? true : false,
+        checkedUsers: checkedUsers.length > 0 ? checkedUsers : [],
+      });
+    }
+  }
+
+  render() {
+    const { users, searchValue, selectedUsers, checkedUsers, isDropdownVisible, isSubmitted } = this.state;
+    let dropDownUsersList = this.transformDropdownUsers();
+    return (
+      <div className={isDropdownVisible ? "overlay" : ''} onClick={this.closeDropdownOutsideClick}>
+        <div className="d-flex users-detail-container mtop-30">
+          <div className="search-container">
+            <input type="text" className="search" placeholder="Search user" value={searchValue} onChange={this.searchUser} onClick={this.toggleDropdown}></input>
+            {isDropdownVisible &&
```

Comment:
This Dropdown Component TO be separate Component. Its to be controlled only by toggle flga